src/OrleansRuntime/Streams/QueueBalancer/DeploymentBasedQueueBalancer.cs

```diff
@@ -98,19 +98,33 @@ public class DeploymentBasedQueueBalancer : QueueBalancerBase, ISiloStatusListen
             await NotifyListeners();
         }
 
+        private async Task HandleSiloStatusChange()
+        {
+            while (!this.siloStatusOracle.CurrentStatus.IsTerminating())
+            {
+                try
+                {
+                    this.siloStatusSharedState = await this.siloStatusSharedState.NextAsync;
+                    HandleSiloStatusChange(this.siloStatusSharedState.State);
```

Comment:
> Think ideal behavior is only process new state when there's a change

A new state is only available for processing when the task on NextAsync returns the next shared state.  The task in the next shared state is not resolved until a new state is published, so upon the task of NextAsync being resolved, we have a new state to process, and a new task to await for the next state change.